packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -92,11 +93,14 @@ export async function fetchServerResponse(
   }
 
   /**
-   * Three cases:
+   * Four cases:
    * - `prefetchKind` is `undefined`, it means it's a normal navigation, so we want to prefetch the page data fully
+   * - `prefetchKind` is `temporary` - it means it's a normal navigation, so we want to prefetch the page data fully
    * - `prefetchKind` is `full` - we want to prefetch the whole page so same as above
    * - `prefetchKind` is `auto` - if the page is dynamic, prefetch the page data partially, if static prefetch the page data fully
    */
+  const isNavigation =
```

Comment:
>for example router.refresh will call this function without a prefetch kind. 

That is the terminology that used in the existing comment:
> `prefetchKind` is `undefined`, it means it's a normal navigation

> Is there significance to gating this logic behind the prefetchKind value or would it make sense to always do this check in development?

I thought that's what we originally discussed? https://github.com/vercel/next.js/pull/67673#discussion_r1674372468

I can remove this gating but this whole block seems a bit problematic since the comment seems to no longer reflect it's functionality. Or maybe it's just the wording that needs updating.

Reply:
I've reverted the changes surrounding `isNavigating` for now until we can clarify the wording around `prefetchKind`